### PR TITLE
modify existing validators() rust api to take both epoch_id and block_id

### PR DESCRIPTION
**Pull request by ppca**
1. removed the previously added `validator_by_epoch_id()` api
2. modified the pre-existing `validators(maybe_block_id)` api to be `validators(maybe_epoch_reference)`. Basically now within rust, we can input `None`, or `Some(EpochReference::EpochId(epoch_id))` or `Some(EpochReference::BlockId(block_id))`. 
3. added some unit tests to test all currently supported json inputs to the `validators` RPC endpoint are working. When users send requests to the near rpc validators endpoint, they could use as input the legacy format `[block_id]` or the map format `{"epoch_id": epoch_hash}` or `{"block_id": block hash/height}`

### CODE REVIEW

File: core/primitives/src/types.rs
Comment (nikurt) [inline core/primitives/src/types.rs:908]: I'm not in favor of defining types for `Option`s. The resulting code is almost the same length, but introduces an additional concept.

File: chain/jsonrpc/src/api/validator.rs
Comment (nikurt) [inline chain/jsonrpc/src/api/validator.rs:72]: You don't need this assert, if you do an `unwrap()` later.

File: chain/jsonrpc/src/api/validator.rs
Comment (nikurt) [inline chain/jsonrpc/src/api/validator.rs:79]: This works, but the code is much simpler if you define `Eq` and `PartialEq` for `EpochReference`. Then you can simply do `assert_eq!(result.unwrap(), RpcValidatorRequest { epoch_reference: EpochReference::BlockId(BlockId::Hash(block_hash)) };`

I think it's acceptable because all network messages define `Eq` and `PartialEq`. `RpcValidatorRequest` is almost a network message. :)

File: chain/jsonrpc/src/api/validator.rs
Comment (nikurt) [inline chain/jsonrpc/src/api/validator.rs:70]: Remove `block_hash_string` and do this instead

```suggestion
        let params = serde_json::json!([block_hash.to_string()]);
```

File: chain/jsonrpc/src/api/validator.rs
Comment (telezhnaya) [inline chain/jsonrpc/src/api/validator.rs:101]: Very nice, it works the same as we have in blocks API (the ability of parsing both height and hash under the same name)
https://docs.near.org/api/rpc/block-chunk#block-details

Could you please update docs repo as well? Mentioing that we have a new format now
Be careful and don't merge it until we release 1.37

Comment (ppca) [reply chain/jsonrpc/src/api/validator.rs:101]: Hey Olga, what is the docs repo you are referring to?
And when you say don't merge it until we release 1.37, do you mean the change to the docs repo for the api change?

Comment (telezhnaya) [reply chain/jsonrpc/src/api/validator.rs:89]: https://github.com/near/docs

> do you mean the change to the docs repo

yes ofk
It's because the changes in nearcore will be live only after the release of 1.37, while the docs will be updated right after the merge
My PR is also waiting https://github.com/near/docs/pull/1564

Comment (ppca) [reply chain/jsonrpc/src/api/validator.rs:89]: Got it! Will open a PR in docs tmr.

**Review by nikurt — APPROVED** [review]
Looks good.
And :+1: for the concise tests.

**Pull request merged**
